agent/src/test/java/com/teamscale/jacoco/agent/store/upload/delay/DelayedCommitDescriptorRetrievalTest.java

```diff
@@ -16,23 +18,25 @@
 	@Test
 	public void locatorShouldTriggerUploadOfCachedXmls() throws Exception {
 		ExecutorService storeExecutor = Executors.newSingleThreadExecutor();
-		InMemoryStore cache = new InMemoryStore();
-		InMemoryStore destination = new InMemoryStore();
-		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, cache,
+		File coverageFile = File.createTempFile("jacoco-", ".xml");
+		Path outputPath = coverageFile.getParentFile().toPath();
+
+		InMemoryUploader destination = new InMemoryUploader();
+		DelayedCommitDescriptorStore store = new DelayedCommitDescriptorStore(commit -> destination, outputPath,
 				storeExecutor);
 
 		ExecutorService locatorExecutor = Executors.newSingleThreadExecutor();
 		GitPropertiesLocator locator = new GitPropertiesLocator(store, locatorExecutor);
 
-		store.store("xml1");
+		store.upload(coverageFile);
 		locator.searchJarFileForGitPropertiesAsync(new File(getClass().getResource("git-properties.jar").toURI()));
 		locatorExecutor.shutdown();
 		locatorExecutor.awaitTermination(5, TimeUnit.SECONDS);
 		storeExecutor.shutdown();
 		storeExecutor.awaitTermination(5, TimeUnit.SECONDS);
 
-		assertThat(cache.getXmls()).isEmpty();
-		assertThat(destination.getXmls()).containsExactly("xml1");
+		assertThat(Files.list(outputPath).anyMatch(path -> path.getFileName().equals(coverageFile.toPath())))
```

Comment:
is this assertion really correct? you're filtering for paths where the file name matches the entire path of the coverage file. I'd wager that is never true in any circumstances. shouldn't the `equals` just compare both paths (i.e. without the `getFilename`) or both file names (i.e. add a `getFileName to the right hand side)? can you please try and provoke a test failure to check that the assertion actually works as intended?

Reply:
yeah, that's a mistake, removed the `getFileName` call.